eclipse/src/saros/ui/views/SarosView.java

```diff
@@ -456,7 +457,13 @@ public void menuAboutToShow(final IMenuManager manager) {
                 SelectionRetrieverFactory.getSelectionRetriever(JID.class).getSelection();
             if (contacts.size() == 0) return;
 
-            if (sarosSessionManager.getSession() == null) {
+            final JID jid = contacts.get(0);
+
+            // FIXME dirty hack
+            final ContactPropertyTester tester = new ContactPropertyTester();
```

Comment:
I am not completely sure on how the tester should be used properly, but, as also indicated by the "dirty hack" fixme, this does not seem to be the correct way.
Are there any restrictions keeping us from using the tester correctly? (And how are they supposed to be used anyways.? The javadoc of `PropertyTester` doesn't really seem helpful to me...)

Reply:
We got an issue with Eclipse 4. Maybe you look at my former commits. The entry that is displayed here was handle be the magic XML stuff in the plugin.xml. With Eclipse 4 it broke and so we have to do it manually.

Reply:
Better explanation. Suppose you have a JTree. If you right - click on on entry context menu will appear, basically the logic here is building the context menu. But with Eclipse it (was) is possible to even append this menu, e.g when you open the context menu for a tree item your own logic ads A,B,C and on top of that the logic defined in the plugin.xml will add D,E, F.

Reply:
Ok, so the whole config stuff in the plugin.xml was also added in case this check ever were to function again in the future? Otherwise, if the logic is broken and unnecessarily complicated to use, why add the config for it?

Reply:
Are you aware that the logic in the plugin.xml adds the context menu item "Add contact to session" and the logic in the SarosView adds "Share Projects..." ?

Reply:
No. From your explanations, I just assumed that everything dealing with the plugin.xml menu stuff was broken. But if some stuff is still working, the question basically answers itself.

How about amending the FIXME addressing how/when to fix/replace the added hack?